src/Repository/PostRepository.php

```diff
@@ -64,7 +64,7 @@ private function createPaginator(Query $query, int $page): Pagerfanta
     /**
      * @return Post[]
      */
-    public function findBySearchQuery(string $rawQuery, int $limit = Post::NUM_ITEMS): array
+    public function findBySearchQuery(string $rawQuery, int $limit = Post::NUM_ITEMS): iterable
```

Comment:
nope. Doctrine will always use an array there

Reply:
Mmm ok.  
In which cases do we get an `ArrayCollection` ?

Reply:
In case of calling getters for properties which have OneToMany/ManyToMany relations, e.g. `$post->getTags()`, etc. You can look for properties which have `@var ArrayCollection` annotation.